pkg/controller/operators/catalog/operator.go

```diff
@@ -1424,49 +1426,52 @@ func (o *Operator) syncInstallPlans(obj interface{}) (syncError error) {
 }
 
 type installPlanTransitioner interface {
-	ResolvePlan(*v1alpha1.InstallPlan) error
 	ExecutePlan(*v1alpha1.InstallPlan) error
 }
 
 var _ installPlanTransitioner = &Operator{}
 
-func transitionInstallPlanState(log *logrus.Logger, transitioner installPlanTransitioner, in v1alpha1.InstallPlan, now metav1.Time) (*v1alpha1.InstallPlan, error) {
+func transitionInstallPlanState(log logrus.FieldLogger, transitioner installPlanTransitioner, in v1alpha1.InstallPlan, now metav1.Time, timeout time.Duration) (*v1alpha1.InstallPlan, error) {
 	out := in.DeepCopy()
 
 	switch in.Status.Phase {
 	case v1alpha1.InstallPlanPhaseRequiresApproval:
 		if out.Spec.Approved {
-			log.Debugf("approved, setting to %s", v1alpha1.InstallPlanPhasePlanning)
 			out.Status.Phase = v1alpha1.InstallPlanPhaseInstalling
+			out.Status.Message = ""
+			log.Debugf("approved, setting to %s", out.Status.Phase)
 		} else {
 			log.Debug("not approved, skipping sync")
 		}
 		return out, nil
 
 	case v1alpha1.InstallPlanPhaseInstalling:
+		if out.Status.StartTime == nil {
+			out.Status.StartTime = &now
+		}
 		log.Debug("attempting to install")
 		if err := transitioner.ExecutePlan(out); err != nil {
-			out.Status.SetCondition(v1alpha1.ConditionFailed(v1alpha1.InstallPlanInstalled,
-				v1alpha1.InstallPlanReasonComponentFailed, err.Error(), &now))
-			out.Status.Phase = v1alpha1.InstallPlanPhaseFailed
+			if now.Sub(out.Status.StartTime.Time) >= timeout {
```

Comment:
I know we are extremely pressed on time here but I envisioned that we would actually check the error type as well. For example if it's a 409 or something, we certainly retry. If it's "a CSV with the same name already exists" then we don't. But it's not worth trying to implement some kind of retriable error function that returns a bool?

Reply:
I figured we would do the opposite, since plan execution itself is supposed to be idempotent, and eventually tag specific error cases that definitely won't resolve with retries. But I think that's a future enhancement, since we don't have a complete list.

Reply:
It's definitely something for the future, yes. I see what you mean about doing the opposite. Instead of catching errors that are requauable (which can be large in number) track those errors which are definitely terminal and fail-fast on those. Makes sense. 

At my last company we started trying to build something like an `IsRequeuable(error) bool` type thing, not sure it really went anywhere